src/netaddress.cpp

```diff
@@ -287,17 +283,12 @@ bool CNetAddr::IsValid() const
     if (IsInternal())
         return false;
 
-    if (IsIPv4())
-    {
-        // INADDR_NONE
-        uint32_t ipNone = INADDR_NONE;
-        if (memcmp(ip+12, &ipNone, 4) == 0)
-            return false;
-
-        // 0
-        ipNone = 0;
-        if (memcmp(ip+12, &ipNone, 4) == 0)
-            return false;
+    if (IsIPv4()) {
+        for (uint32_t a : {(uint32_t)INADDR_ANY, (uint32_t)INADDR_NONE}) {
```

Comment:
Not something you introduced, but this only works because those addresses have the same byte representation in every byte order.

Reply:
Right! Even though the values of those constants is never going to change (I hope!), does the following look better?

```cpp
for (uint32_t a : {htonl(INADDR_ANY), htonl(INADDR_NONE)}) {
```

Reply:
That seems the best option, I think. It's too bad that there's no way to get these constants as byte spans that can be directly compared against m_addr so it wouldn't need the cast and `memcmp` at all.

Reply:
Changed to use `htonl()`.